server/src/main/java/org/apache/iotdb/db/metadata/rescon/MemSchemaEngineStatistics.java

```diff
@@ -69,19 +69,27 @@ public long getMemoryUsage() {
   public void requestMemory(long size) {
     memoryUsage.addAndGet(size);
     if (memoryUsage.get() >= memoryCapacity) {
-      logger.warn("Current series memory {} is too large...", memoryUsage);
-      allowToCreateNewSeries = false;
+      synchronized (this) {
+        if (allowToCreateNewSeries && memoryUsage.get() >= memoryCapacity) {
+          logger.warn("Current series memory {} is too large...", memoryUsage);
+          allowToCreateNewSeries = false;
+        }
+      }
     }
   }
 
   public void releaseMemory(long size) {
     memoryUsage.addAndGet(-size);
-    if (!allowToCreateNewSeries && memoryUsage.get() < memoryCapacity) {
-      logger.info(
-          "Current series memory {} come back to normal level, total series number is {}.",
-          memoryUsage,
-          totalSeriesNumber);
-      allowToCreateNewSeries = true;
+    if (memoryUsage.get() < memoryCapacity) {
+      synchronized (this) {
```

Comment:
```suggestion
      synchronized (allowToCreateNewSeriesLock) {
```

It is better to use a new object rather than this

Reply:
Fixed.